### DISCUSSION TITLE
Passing an array literal to contains() function causes syntax error

### DISCUSSION THREAD

**Discussion by lots0logs** [community/community#27223] [Actions]
<p>Not sure if this is a bug or if I am doing it wrong?</p>
<pre><code>- name: Register/Restore Node Modules Cache - EPanel
  if: false == contains(['monarch', 'bloom'], github.event.repository.name)
</code></pre>
<p>Result:</p>
<pre><code>### ERRORED 18:40:27Z

- Your workflow file was invalid: The pipeline is not valid. .github/workflows/main.yml (Line: 49, Col: 11): Unexpected symbol: '['. Located at position 19 within expression: false == contains(['monarch', 'bloom'],
</code></pre>

**Comment by ericsciple** [comment, marked as answer]
<p>Unfortunately the expression syntax doesn’t support JSON. An array would have to dereferenced from some object in the context - e.g. github.event.my_array</p>
<p>The current solution for the specific scenario is <code>if github.event.repository.name == 'monarch' || github.event.repository.name == 'bloom'</code></p>

**Comment by dzervas** [comment]
<p>There’s also the ability to do it as follows:<br>
<code>contains(fromJson('["monarch", "bloom"]'), github.event.repository.name)</code></p>

**Reply by rvesse** [reply]
This is a much better solution if you have more than a couple of acceptable values, thanks 👍
